### ISSUE TITLE
Everybody agrees to substitute Friend concept to Trusted node concept?

### ISSUE THREAD

**Issue by pedro-nonfree** [RetroShare/RetroShare#1418]
I don't know if you all agree that friend concept is not correct for the retroshare and GXS ecosystem.

In PGP/GPG it is clear that it is a matter of trust (and this is probably the same for retroshare)

I would like to see a +1 before starting a massive substitution in code. The idea for this issue is to close the issue when the concept is fully migrated. If we do everything in one commit it is easier to revert and inspect. The idea is also to automate the substitution so it is easy for everybody to apply to current developing branches, etc.

The proposal is to substitute:

text:

Friend -> Trusted Node

code:

friend -> trustedNode

**Comment by csoler** [comment]
In the UI we tried to substitute "Friend" for "trusted node", since that is the correct idea behind connections, but there's still a few places where the old "Friend" term remains.

Generally speaking, massive substitutions in the code are not accepted.

So in summary, it's ok for you to make a pull request to change "Friend" for "Trusted node" in the UI.

**Comment by pedro-nonfree** [comment]
@csoler so you probably prefer substitute little by little (small commits) parts of user interface, right?

**Comment by pedro-nonfree** [comment]
I commented with @G10h4ck and I will go by parts (yay! I started!)

**Comment by felisucoibi** [comment]
I prefer friend, is more newbies concept.

**Comment by felisucoibi** [comment]
nobody knows what is a trusted node.

**Comment by pedro-nonfree** [comment]
From wikipedia, [friendship](https://en.wikipedia.org/wiki/Friendship):

> Such characteristics include affection; kindness, love, virtue, sympathy, empathy, honesty, altruism, loyalty, mutual understanding and compassion, enjoyment of each other's company, **trust**, and the ability to be oneself, express one's feelings to others, and make mistakes without fear of judgment from the friend.

and this relationships evolves with age (1.1 Childhood, 1.2 Adolescence, 1.3 Adulthood, 1.4 Older adults)

Hence, a friend is a set of a lot of different characteristics that depends on age. Maybe: culture,  sexual orientation (and/or the way you love), religion, politics, etc.

Trust is a thing that may be related to a friend. A friend that you don't trust you may call him your friend. Someone you trust is probably not your friend, for example the guy that is storing your money (and you see, sometimes is an infrastructure, a trademark)... then, if we put retroshare locations, that are not friends.

So we are helping these newbies comming from fake social networks, but we are destroying the concept retroshare wants to put on the table. That fake social networks, for example, don't want people to make understand privacy, security. That's why for them is difficult to have something to hide, or to understand what is a node, what is a network. Hey! They don't understand that because that fake networks (plus related technologies) focus on their narcicism-individualism and not the collective-network thing.

... and we are putting obstacles ourselves the way the network works.

There are two scenarios of confusion:

1. "I add friends like facebook, then I have thousands of them"
2. "I have very few friends, and nearly none of them use retroshare".

Both scenarios are incorrect. The first one is dangerous because your ~~friends~~ trusted nodes can eavesdrop you; the seconds one makes retroshare not usable.

The correct scenario is to think who I trust to relay my connections, in a more "professional" or "neutral" relation.

I think trust is clear that should be there.

"Node" is a problem. But to learn that a network is composed by nodes (entities, circles) and links (relations, lines) is easy to understand and... very useful!

**Comment by felisucoibi** [comment]
I'm so sorry but it makes things more complicate. Friends is OK, strange names make people confuse of what they are used to.
I don't care about what platon thinks, or shopenhauer, thinks, or is even more realistic to use trusted nodes instead of friends is better because it reaches what it really is.

As i said always i'm nobody, i just helped lot of times to make rs more simple and usable... this is only why sometimes i expect people to understand me or at least trust my opinion,  Friends was the original name rs creator used in 2006, after that, somebody decide to change it to trusted nodes in 2009 or 10, after my simplification proposal rs names got again more or less the same they where in 2006, and now again somebody wants to complicate it...

**Comment by defnax** [comment]
i dont agree it.
Friends must stay as friends!
why the hell we must use ugly naming things?
In the past i used simple names, why we must go now complicated?

We need make retroshare userfriendly and easy and but now started all with renaming trusted nodes stuff.
For the few rs geeks/expert users ?

but why not making adding/find friends easyer? with short keys instead of ugly naming since few years

**Comment by csoler** [comment]
Let me backup a little bit:
- contact: GXS identities that are in your contact list.
- friend nodes: RS nodes that you allow your own RS node to connect to. 
- profiles: PGP key that authenticate RS nodes. As you know, multiple nodes can share the same key.

So in this context, "friends" is a very ambiguous term that may confuse users. I agree that the idea of friendship is good, so I vote for "Friend nodes". 

The PGP layer is really about trust: that's where you allow to connect or not. Because we have GXS identities, we will sooner or later remove the chat between nodes so as to only keep the chat between identities. It will be much less confusing. In this perspective, nodes can really be seen as computers running RS instances.

I don't like "profiles" too much either. Among options, we have:
1 - contact / friend / friend: what we had before (defnax's good old times ;-) )
2 - contact / trusted node / profile: what we have now. 
3 - contact / node / trusted friends: this keeps friend=PGP key, and let "node" be one of the machines the friend runs. 
4 - contact / computer / profile: insists on the fact that the real people are to be found as GXS identities
5 - friend / computer / profile: even better.

I prefer 5. I'd be glad to hear everyone's thoughts.

**Comment by defnax** [comment]
yes im for 4 or 5👍

i never liked trusted & node words

**Comment by pedro-nonfree** [comment]
### simple mode

I'm for 4 @csoler 

### discussion mode

- replace People / *My contacts* with *Friends*
- Put *Online Friends/Total Friends* as the number of online contacts in the category of *Friends* (ex *My contacts*)
- Add *Online computers/Total computers* between the status bar elements *Online Friends/Total Friends* and *NAT*
- Computers is nice, from there I think Servers is better (is it easy to understand for users "In retroshare, you connect to servers and you run your own server too"?). Another possibility: direct connections (vs dht connections).
    - In patchwork, they solved it with the concept of [pub](https://en.wikipedia.org/wiki/Pub):
        - https://github.com/ssbc/patchwork#pubs
        - https://github.com/ssbc/scuttlebot/wiki/Pub-Servers
- Profile: profile is a word more used in "set of configurations", "preferences", customizations by many programs. With this part I get very confused about how retroshare works so I'm very limited to contribute. Let's try. Given that there are "locations" where you store profiles, probably is better to store *keys* or *certificates* that are words more related to *verify*, *authenticate*, *secure* communications (words not only valid for computers, but also for 17th century.)

Probably, a summary would be: contact (friend subset) / server / certificate

### deeper discussion mode

Using definitions of people and friends are we participating in this tendency of hiding users and computers? http://contemporary-home-computing.org/turing-complete-user/

**Comment by csoler** [comment]
contact (friend subset) / server / certificate  sounds very good to me.

**Comment by felisucoibi** [comment]
How many wasted time now that there are lot of things to do. But my proposal....

if
    contact: GXS identities that are in your contact list.
    friend nodes: RS nodes that you allow your own RS node to connect to.
    profiles: PGP key that authenticate RS nodes. As you know, multiple nodes can share the same key.
then
Contact should be PEOPLE
Trusted nodes should be FRIENDS
Profiles should be COMPUTER or PLACE or MACHINE

**Comment by defnax** [comment]
yes i like your idea the best one @felisucoibi 👍
cyrils one goes more technical naming

1. **People** (gxs profiles)) you talk over the rs network without need to connected. 
2. **Friends** your direct contacts
3. **Profiles** pc, notebook, location or else

**Comment by pedro-nonfree** [comment]
Just a note, seems is not clear why I opened this issue. I'm happy with any definition that helps me to understand how retroshare works (to a technician and to a basic user). Let's put coherence in the UI. I want to help on that.

I don't know if you are fine explaining thing number 1 to users and thing number 2 to technicians. I did not see this approach in any other technology I'm using. Definitions should be unified between all diverse users that increasingly understand and participate the platform.

> profiles: PGP key that authenticate RS nodes. As you know, multiple nodes can share the same key.

translates to:

profiles: computer that authenticate friends. As you know, multiple friends can share the same computer.

For me this sentence is not working (can you help me?), same for: pc, notebook, location, machine.

I don't know what do you think about:

profiles: certificate that authenticate servers. As you know, multiple servers can share the same certificate.

... server might allocate a contact (gxs identity) that can be your friend - interface can put friend thing very present to help users to contact them

certificate is what makes connections secure. I think certificate term is very clear for lots of users/people: HTTPS certificate, personal certificates in workplace or individual usage, and classic certificates in paper and seal.

The language we use should explain the things retroshare is capable of.

In the time of [napster](https://en.wikipedia.org/wiki/Napster#/media/File:Napster_running_on_an_original_iBook_(2001-03-11).png) revolution, it succeeded with the word of servers. Same for emule (that was more present in my circles).

We want to socialize that, no problem. But why we have to hide the server thing with fake concepts that don't help to understand the architecture.

And please, avoid using unpleasant language, am I unpleasant to you?

**Comment by sehraf** [comment]
> profiles: computer that authenticate friends. As you know, multiple friends can share the same computer.

indeed. 
> Trusted nodes should be FRIENDS
> Profiles should be COMPUTER or PLACE or MACHINE

is nonsense since a profile can be the same for multiple computers

**Comment by felisucoibi** [comment]
"We want to socialize that, no problem. But why we have to hide the server thing with fake concepts that don't help to understand the architecture."
Since when user-side needs to understand the architecture? why somebody needs to udnerstand how weight is balanced in a builidng to enter in to it and buy some potatoes?
Sorry for my language, i'm kind of simple and uneducated.

**Comment by defnax** [comment]
the best was when we have only one profile like other software, example skype, tox, ring

 Friends (direct) & People (none friend person in rs net) 

my  rsprofile has by default a gxs id for communication with rs net people all in one profile in future would be bether as 3 things.

i dont know why in the past one developer added this ssl location stuff on retroshare.

**RetroShare 0.4** was the easiest retroshare where we had
sorry for offtopic but we can have so easy all

**Comment by pedro-nonfree** [comment]
Trying again with a mix between the old and new things.

A friend is composed by:

- *friend person* (plural people): a GXS **identity** that is related to a *friend server* and that is listed in people (contact list).
    - *person* (plural people): a GXS identity that is related to a **server (indirect connection)** or pseudoanonymous (unknown server) and that is listed in people (contact list).
- *friend server*: **direct connection** to a peer you trust 
- *friend certificate*: PGP key associated to a *friend server* that guarantees **security** (authenticity and/or encryption) between you and your *friend server*. Multiple friend server can share the same friend certificate, this have two use cases: (1) your friend uses different servers or devices, (2) your friend is a shared *friend person* used by some of your friends

Then, *friend* involves the "only one profile thing", the simple case; i.e. someone that owns one *friend server* with one *friend certificate* and one *friend person*. Just a friend.

The people that wants to decouple the concepts **identity, direct connection and security** to benefit from a special use case must specify clearly the friend subcomponent that is using: person, server, certificate.

**Comment by csoler** [comment]
I agree with person/server/certificate. One details though: "server" sounds like a big computer running lots of connections. It practice (that means everytime except in very rare situations) it's just a computer. So I would preferably go for "person/computer/certificate", but I would still go for person/server/certificate. 

At least, it's way better than person/server/friend, that is really confusing.

**Comment by felisucoibi** [comment]
if
contact: GXS identities that are in your contact list.
Friend nodes: RS nodes that you allow your own RS node to connect to.
profiles: PGP key that authenticate RS nodes. As you know, multiple nodes can share the same key.
then
Contact should be PEOPLE
Trusted nodes should be FRIENDS
Profiles should be COMPUTER or PLACE or MACHINE

as alwais i'm nobody, this is my proposal.

**Comment by defnax** [comment]
i agree @felisucoibi he thinks like me use simple words for everyone

- Contacts should be PEOPLE
- Trusted nodes should be FRIENDS
- Profiles should be COMPUTER or PLACE or MACHINE

**Comment by csoler** [comment]
"Profiles should be computer or place": this is nonsense. A given profile (i.e. a PGP key) can have as many locations as it want. So Profile is not a machine. That really makes no sense. If you want to keep the "friends" idea, then
Contacts => Contacts, because People is *every GXS identity* and contact is just the ones you want to keep in your adress book
Trusted nodes => computer. That's what it is.
Profile => Friend.

**Comment by csoler** [comment]
But I really think that Friend should rather be "certificate", because in the end, what you exchange is certificates. So at least, if you call that list certificates, you know what's in it.

**Comment by defnax** [comment]
People (for everyone in rs net, )
Contacts ( which are your favorites special person in rs network)
Friends you direct  trustedfriends

importend is for me friends need to stay as friends not a technical naming please

**Comment by pedro-nonfree** [comment]
@defnax @felisucoibi I would like to know what do you think about my previous proposal taking in account your comments (preserving friend and simple as possible): https://github.com/RetroShare/RetroShare/issues/1418#issuecomment-445416955

When you have a friend is in the contacts/people tab, you have a connection to him (server, computer) and you have a certificate. So is your friend appearing in these three different scopes: as a contact, as a connection and his certificate/seal/guarantee/verification. So it is *just a friend*, but then I continued being concrete and compatible with the technical part (friend person, friend server, friend certificate)

**Comment by felisucoibi** [comment]
> But I really think that Friend should rather be "certificate", because in the end, what you exchange is certificates. So at least, if you call that list certificates, you know what's in it.

when you add somebody in whatsapp you add a certificate and you use secure e2e connections... but nobody of the 1.000.000.000 users por whatsapop ever heard of it....

**Comment by csoler** [comment]
Well, we're not supposed to copy whatsapp either. Personnaly I prefer that users of RS understand the basics of security model they're using, even if it means less users, because it prevents them to make potentially critical mistakes and therefore improves security a lot.

**Comment by felisucoibi** [comment]
For me now, the most important thing i need in rs is faster GUI in FILES, more responsive in search results and flielists browsing, this is the most important for me now, that still makes rs useless to some users because filelist browsing is slow, I will no talk more about this topic, do what you consider is better for RS, I'm not angry, i dont want also to make you guys angry, it's just too much time for something not iportant at all atm for me.

I proposed this:
https://github.com/RetroShare/RetroShare/issues/1018
1. FILELISTS

File list in rs are not very good for lot of files, they are slow to browse, complicated to use when there are lot of files in the same directory, lot of memory usage, hangs... We can learn a lot of dc++ they took years to decide the best aproach for file lists browsing. I wrote something about here:

[GUI STUFF] Improvements in file lists. retroshare://forum?name=%5BGUI%20STUFF%5D%20Improvements%20in%20file%20lists.&id=53ee154e19536550c35b507e2c67167a&msgid=d91fe54a67eac0beae0a184ac9a7447e198665cd

In dc++ they use 3 columns, first one to list users with file lists, the second column is for the tree directory, and the third for files and directories inside a directory.
These 3 levels allow easy to change between one user and another, not load all the files in the same list, the second column is only for directories, and the third column contains files of the selected directory, and subdiretories if there are. This also allows to not necessary load all the files and directories in the same tree probably makes things faster and responsive, not necessary to load 100K files all the time, just when yu open the directory and just to show them in the third column.
And also these 3 levels allow to browse files with arrows of the keyboard, up down to move up and down and left right to enter in to a directory or exit from it.

File lists now in RS with big ones, are slow, unresponsive, and complicated to manage if lot of files inside a directory, you need to move your self up and down just to close a opened directory with lot fo files, as more files as more you have to move up and down etc.

**Comment by csoler** [comment]
@felisucoibi; that is a nice idea, but it requires some coding: modifying the file model etc. Can't do that for 0.6.5, or we will postpone the release of 1 more month. Definitely not possible. 
I think that what we need is to offer the 3-way view that you suggest as an option (like flat view, threaded view, 3-way view). It's only useful for people who share a huge number of files I suppose.
How would you do the search in such a view? It's practically not possible btw.

**Comment by felisucoibi** [comment]
> @felisucoibi; that is a nice idea, but it requires some coding: modifying the file model etc. Can't do that for 0.6.5, or we will postpone the release of 1 more month. Definitely not possible.
> I think that what we need is to offer the 3-way view that you suggest as an option (like flat view, threaded view, 3-way view). It's only useful for people who share a huge number of files I suppose.
> How would you do the search in such a view? It's practically not possible btw.

Serches in this mode are made in normal search, maybe with only friend filelists marked, it's ok if is nto posible to search in 3 column way, if we get lot of speed browsing and simplify filelists browsig.

Is not only usefull for people that shares huge filelists, filelists browsing in tree mode and flat mode are broken, more flat than normal, but both are broken, they are very very slow, also search in filelists.

**Comment by defnax** [comment]
@csoler close?
